src/app/tests/ember-test-compatibility.cpp

```diff
@@ -0,0 +1,127 @@
+#include "lib/support/CHIPMem.h"
+#include <app/tests/ember-test-compatibility.h>
+#include <app/util/basic-types.h>
+
+#include <app/InteractionModelEngine.h>
+#include <app/MessageDef/AttributeReportIBs.h>
+#include <app/util/basic-types.h>
+#include <app/util/mock/Constants.h>
+#include <app/util/mock/Functions.h>
+#include <lib/core/CHIPCore.h>
+#include <messaging/ReliableMessageContext.h>
+
+namespace chip {
+uint8_t chip::Test::attributeDataTLV[CHIP_CONFIG_DEFAULT_UDP_MTU_SIZE];
+size_t chip::Test::attributeDataTLVLen = 0;
+namespace app {
+
+// was previously in TestWriteInteraction.cpp
+const EmberAfAttributeMetadata * GetAttributeMetadata(const ConcreteAttributePath & aConcreteClusterPath)
+{
+    // Note: This test does not make use of the real attribute metadata.
+    static EmberAfAttributeMetadata stub = { .defaultValue = EmberAfDefaultOrMinMaxAttributeValue(uint32_t(0)) };
+    return &stub;
+}
+
+// was previously in TestWriteInteraction.cpp
+CHIP_ERROR WriteSingleClusterData(const Access::SubjectDescriptor & aSubjectDescriptor, const ConcreteDataAttributePath & aPath,
```

Comment:
This has nothing to do with "ember".  Same for a lot of the other functions being added here.

Reply:
if I understood correctly, the original form of this function (the one defined in `ember-compatibility-functions.cpp`) interacts directly with ember layer. and this is a stub definition to override it, no?
 what do you suggest I call it?

Reply:
I suggest you call this `test-interaction-model-api.cpp` or something like that.  `WriteSingleClusterData` is a backend-agnostic API that the IM uses.  The implementation in ember-compatibility implements `WriteSingleClusterData` on top of ember bits, but your implementation has nothing to do with ember....

Reply:
FYI, it was suggested by @bzbarsky-apple to create another file for the ember specific bit: since I just added a new stub `emberAfGetClusterServerEndpointIndex()` which is in fact more ember related